services/security.ts

```
@@ -15,3 +15,14 @@ export const isValidUrl = (url: string): boolean => {
 export const sanitizeUrl = (url: string): string => {
   return isValidUrl(url) ? url : '';
 };
+
+export const sanitizeInput = (input: string, maxLength: number = 1000): string => {
+  if (!input) return '';
+  // Remove control characters (0-31, 127) except tab, newline, carriage return
+  let sanitized = input.replace(/[\x00-\x08\x0B\x0C\x0E-\x1F\x7F]/g, '');
```

Biome (2.3.13)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)